go/vt/topo/tablet.go

```diff
@@ -263,13 +266,26 @@ func (ts *Server) GetTabletsByCell(ctx context.Context, cellAlias string, opt *G
 		return nil, err
 	}
 
-	tablets := make([]*TabletInfo, len(listResults))
+	var capHint int
+	if opt != nil && opt.KeyspaceShard == nil {
+		capHint = len(listResults)
+	}
+
+	tablets := make([]*TabletInfo, 0, capHint)
 	for n := range listResults {
 		tablet := &topodatapb.Tablet{}
 		if err := tablet.UnmarshalVT(listResults[n].Value); err != nil {
 			return nil, err
 		}
-		tablets[n] = &TabletInfo{Tablet: tablet, version: listResults[n].Version}
+		if opt != nil && opt.KeyspaceShard != nil {
+			if opt.KeyspaceShard.Keyspace != "" && opt.KeyspaceShard.Keyspace != tablet.Keyspace {
+				continue
+			}
+			if opt.KeyspaceShard.Shard != "" && opt.KeyspaceShard.Shard != tablet.Shard {
+				continue
+			}
```

Comment:
So you can pass in an empty keyspace but a shard of "-" e.g. to match tablets in any unsharded keyspace. Is that intentional? I'm not sure it's wrong, but checking as the test currently assumes there's only 1 matching keyspace but here there could be N.

Reply:
If we do want to support that (shard w/o a keyspace), let's add a test case for it.

Reply:
Updated to require a keyspace